.github/workflows/ci.yml

```diff
@@ -36,7 +36,7 @@ jobs:
 
     - name: Analyze with mypy
       run: |
-        python -m mypy
+        python -m mypy --install-types --non-interactive
```

Comment:
mypy tests were failing with:

```
src/gamatrix/__main__.py:29: error: Library stubs not installed for "pkg_resources"  [import]
src/gamatrix/__main__.py:29: note: Hint: "python3 -m pip install types-setuptools"
src/gamatrix/__main__.py:29: note: (or run "mypy --install-types" to install all missing stub packages)
src/gamatrix/__main__.py:29: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
```

There doesn't seem to be a way to add command-line args to mypy in the toml file, so I had to add them here.

Reply:
This works just fine. I'll play around with this later and see if I can improve things. Also, I'll take a closer look at Ruff as it seems simpler. (There is also a VSCode plugin for it so local configuration should be a bit easier).

https://mypy.readthedocs.io/en/stable/config_file.html#example-pyproject-toml